From b6e8c77c32ab97346f42f12fb8c38461856ffd97 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 27 Jul 2026 18:06:02 +0800
Subject: [PATCH 333/334] AOSCOS: Loongson64: env: fix build failure by adding
 include to <linux/string.h>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When building v7.2-rc5 on MIPS, the following error occurs:

  arch/mips/loongson64/env.c: In function ‘prom_lefi_init_env’:
  arch/mips/loongson64/env.c:272:17: error: implicit declaration of function ‘strncpy’ [-Wimplicit-function-declaration]
    272 |                 strncpy(cpu_full_name, ecpu->cpuname, sizeof(cpu_full_name));
        |                 ^~~~~~~
  arch/mips/loongson64/env.c:30:1: note: ‘strncpy’ is defined in header ‘<string.h>’; this is probably fixable by adding ‘#include <string.h>’
     29 | #include <workarounds.h>
    +++ |+#include <string.h>
     30 |

Add include to <linux/string.h> to fix build.

Fixes: "BACKPORT: FROMLIST: MIPS: Add __cpu_full_name[] to make CPU names more human-readable"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/loongson64/env.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/loongson64/env.c b/arch/mips/loongson64/env.c
index 46bc2ffac652c..5830fac2012a9 100644
--- a/arch/mips/loongson64/env.c
+++ b/arch/mips/loongson64/env.c
@@ -27,6 +27,7 @@
 #include <boot_param.h>
 #include <builtin_dtbs.h>
 #include <workarounds.h>
+#include <linux/string.h>
 
 #define HOST_BRIDGE_CONFIG_ADDR	((void __iomem *)TO_UNCAC(0x1a000000))
 
-- 
2.55.0

